kb/how-to/rewrite-branch-for-coding-style.md

```diff
@@ -0,0 +1,243 @@
+# Switching to the new code style
+
+In January 2023, Mbed TLS switched to a new coding style, which is now enforced by continuous integration tests. This document explains how to adapt ongoing work. Please read this document if you started work on a branch of Mbed TLS that has the old coding style, and you want to contribute this work to Mbed TLS or you are already doing so.
+
+This concerns both maintained branches of Mbed TLS: `development` and `mbedtls-2.28`.
+
+## Coding style overview
+
+The following table summarizes the old and new coding style (before/after January 2023).
+
+| Aspect | Old | New |
+| ------ | --- | --- |
+| Indentation | 4 spaces | 4 spaces |
+| Indentation of `case` | one level deeper than `switch` | one level deeper than `switch` |
+| Maximum line length | 80 preferred | 80 preferred |
+| **Spacing inside parentheses** | always except `define` and casts: `if( f( x ) )` | **never**: `if (f(x))` |
+| Spacing inside square brackets | no: `a[i]` | no: `a[i]` |
+| **Spacing before opening parenthesis** | no: `if( f( x ) )` | **after keyword or binary operator**: `if (f(x))` |
+| Spacing before opening square bracket | no: `a[i]` | no: `a[i]` |
+| Spacing after unary prefix operator | optional: `!c` or `! c` | **no**: `!c` |
+| Spacing around binary operator | yes: `( x + y ) * z` | yes: `(x + y) * z` |
+| Parentheses around return value | mandatory: `return( x );` | **generally not**: `return x;` |
+| **Single-line if/loop body** | separate line, braces optional | separate line, **braces mandatory** |
+| Opening brace | on its own line | **on the same line** except for functions |
+
+For a more detailed description of the coding style, please refer to the [coding standards](../development/mbedtls-coding-standards.md).
+
+## When to migrate
+
+If you have not yet made a pull request, you can migrate your work at any time.
+
+If you have opened a pull request but review has not yet started on it, please migrate your branch to the new code style before review starts.
+
+If there is an ongoing review of a pull request, please communicate with the reviewer(s) to decide on a good time. Rebasing a branch during review is disruptive, because comments end up detached from the code they're about, so it's generally best not to do it. Typically the best time to migrate the branch is after the reviewer(s) have approved.
+
+## How to migrate
+
+### Requirements
+
+Please have the following tools and files available:
+
+* Git 2.17.0 or newer.
+* Uncrustify 0.75.1. Please note that older or newer versions are likely not to give the desired result.
+* Python 3.6 or newer.
+* A Git worktree containing the branch of Mbed TLS you want to migrate.
+* A local copy of [`mbedtls-rewrite-branch-style`](https://github.com/Mbed-TLS/mbedtls-docs/raw/main/tools/bin/mbedtls-rewrite-branch-style) (preferably the most recent version, since it may have received bug fixes).
```

Comment:
```suggestion
* A local copy of [the branch rewriting script `mbedtls-rewrite-branch-style`](https://github.com/Mbed-TLS/mbedtls-docs/raw/main/tools/bin/mbedtls-rewrite-branch-style) (preferably the most recent version, since it may have received bug fixes).
```
The fact that this is a link is not obvious with the backtick formatting on RTD. Including more words in the link will make it noticeable and make sure noone gets stuck wondering where to get the script from.